### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 20.205.243.168:443"

### DISCUSSION THREAD

**Discussion by wudidaqueer** [community/community#64334] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I installed copilot in vscode, but I can't connect to the server. The error message is GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 20.205.243.168:443". I can open GitHub.com in the browser, and access to other websites is also normal. vscode version is 1.81.1, copilot version is v1.104.341.I have purchased copilot under my github account.I already tried reinstalling copilot also to no avail.

**Comment by rainbow1016** [comment]
very good question!

**Comment by youkei-zzz** [comment]
> ### 选择主题领域
> 问题
> 
> ### 身体
> 我在vscode中安装了copilot，但是无法连接到服务器。错误消息是 GitHub Copilot 无法连接到服务器。扩展激活失败：“连接 ETIMEDOUT 20.205.243.168:443”。我可以在浏览器中打开GitHub.com，访问其他网站也正常。vscode版本是1.81.1，copilot版本是v1.104.341。我在我的github帐户下购买了copilot。我已经尝试重新安装copilot也无济于事。

我也有同样的疑问😢😢😢😭

**Reply by zxy02tf** [reply]
在电脑hosts文件中添加下面内容
140.82.112.25                 alive.github.com
140.82.114.6                  api.github.com
185.199.110.153               assets-cdn.github.com
140.82.113.22                 central.github.com
185.199.108.133               cloud.githubusercontent.com
140.82.113.9                  codeload.github.com
140.82.112.22                 collector.github.com
140.82.112.4                  gist.github.com
52.216.146.155                github-cloud.s3.amazonaws.com
52.217.204.137                github-com.s3.amazonaws.com
54.231.163.81                 github-production-release-asset-2e65be.s3.amazonaws.com
52.217.44.156                 github-production-repository-file-5c1aeb.s3.amazonaws.com
52.216.54.145                 github-production-user-asset-6210df.s3.amazonaws.com
192.0.66.2                    github.blog
140.82.112.4                  github.com
140.82.114.17                 github.community
185.199.109.154               github.githubassets.com
151.101.65.194                github.global.ssl.fastly.net
185.199.110.153               github.io
185.199.110.153               githubstatus.com
140.82.112.25                 live.github.com
13.107.43.16                  pipelines.actions.githubusercontent.com
13.107.238.40                 vscode.dev

**Reply by 11213sdfgsdgsd** [reply]
√

**Reply by levinit** [reply]
真的可以了，我的朋友……

**Reply by haamill** [reply]
哥们确实给力

**Reply by xiaolongCcc** [reply]
我的朋友，超级感谢你！

**Reply by ConditionOnClass** [reply]
厉害啊，我的好朋友！

**Reply by lsnls** [reply]
It works, thank you a lot!!

**Comment by davecheney** [comment]
Hello, I'm sorry you are having difficulties. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Comment by 1229308805** [comment]
I have the same question!

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by 1229308805** [reply]
It could be the reason.I can't ping api.github.com now.  And don't know how to fix it.

**Reply by wudidaqueer** [reply]
modify your hosts file, add "140.82.113.5 api.github.com"

**Comment by zhaojixiang** [comment]
I have the same question too!

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by zhaojixiang** [reply]
Yes, I solved it by configuring the proxy address of vscode as https://127.0.0.1:7890
![image](https://github.com/community/community/assets/32761716/ab74ffb4-5859-42f8-aba1-60c5afa26b15)

**Reply by gxr20201212** [reply]
解决了吗？我也是这个问题，如果解决了分享下

**Reply by thazjswe42700** [reply]
你如果是VSCode 你可以和上图一样 更改下vscode的setting.json中的代理

**Reply by gxr20201212** [reply]
不行呢，代理了报代理连接超时GitHub Copilot could not connect to server. Extension activation failed: "tunneling socket could not be established, cause=connect ECONNREFUSED 127.0.0.1:7890"

**Reply by Haale** [reply]
this works for me.
On the premise that you use some shadowrocket protocal software like clash.

**Reply by Haale** [reply]
你得设置成你自己代理软件的端口

**Comment by SunWineee** [comment, marked as answer]
please modify your hosts file, add "140.82.113.5 api.github.com"

**Reply by gxr20201212** [reply]
这个可以

**Reply by wudidaqueer** [reply]
solved my problem，thank you!!

**Reply by hzzjm** [reply]
solved my problem，thank you very much !!

**Reply by xds95** [reply]
现在不行了，能登陆上账号但是不会有反馈

**Reply by Tz2H** [reply]
Appreciate it mate.

**Reply by RatulBh** [reply]
is that working?
...didn't solve my prblem.. thanks..

**Reply by Y-Sisyphus** [reply]
I want to use copilot in remote server via ssh, which hosts file should I modify? local or remote?

**Reply by blink795** [reply]
i had the same problem... thanks!

**Reply by RunYouCleverBoy** [reply]
Hi, Better but now it shows
[error] request to https://api.github.com/user failed, reason: read ECONNRESET

**Comment by zxy02tf** [comment]
Introduction to English：Add the following configuration to the computer hosts file to solve the problem of login failure and response results
中文介绍：在电脑hosts文件，添加如下配置，即可解决无法登录和响应结果
140.82.112.25                 alive.github.com
140.82.114.6                  api.github.com
185.199.110.153               assets-cdn.github.com
140.82.113.22                 central.github.com
185.199.108.133               cloud.githubusercontent.com
140.82.113.9                  codeload.github.com
140.82.112.22                 collector.github.com
140.82.112.4                  gist.github.com
52.216.146.155                github-cloud.s3.amazonaws.com
52.217.204.137                github-com.s3.amazonaws.com
54.231.163.81                 github-production-release-asset-2e65be.s3.amazonaws.com
52.217.44.156                 github-production-repository-file-5c1aeb.s3.amazonaws.com
52.216.54.145                 github-production-user-asset-6210df.s3.amazonaws.com
192.0.66.2                    github.blog
140.82.112.4                  github.com
140.82.114.17                 github.community
185.199.109.154               github.githubassets.com
151.101.65.194                github.global.ssl.fastly.net
185.199.110.153               github.io
185.199.110.153               githubstatus.com
140.82.112.25                 live.github.com
13.107.43.16                  pipelines.actions.githubusercontent.com
13.107.238.40                 vscode.dev

**Reply by Royom** [reply]
Thanks dude

**Comment by lmliminglm** [comment]
之前我的ubrntu上github copliot是可以用的  突然有一天就不能用了，显示：GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 20.205.243.168:443"这个错误  我该如何解决这个问题呢

**Comment by liupeng0622** [comment]
Initially, I tried the method of modifying the hosts file by adding '140.82.113.5 api.github.com'. 
This prevented Copilot from throwing any errors, but it still didn't function properly as it didn't provide any code suggestions. 

Subsequently, I employed another method: I opened the settings in VSCode, searched for 'http.proxy', and set the proxy address and port. 
This got Copilot working smoothly.

![image](https://github.com/community/community/assets/62160086/14dff4dd-bd69-47e9-9426-4890dc033d36)
